docs/attach.rst

```diff
@@ -0,0 +1,90 @@
+Attaching to a running process
+==============================
+
+Memray allows you to attach to a running process and to observe the allocations
+it performs once you've attached. This doesn't allow you to see where memory
+was allocated before you attached to the process, but it does allow you to
+observe or record future allocations. This can be useful if an application is
+continuing to request more memory than you think it should need, and you want
+to figure out why. It can also be useful for observing the allocation patterns
+of a long running process once the process has already completed any initial
+warm up steps that it needs to do.
+
+Basic Usage
+-----------
+
+The general form of the ``attach`` subcommand is:
+
+.. code:: shell
+
+    memray attach [options] <pid>
+
+The only required argument for ``memray attach`` is a process ID to attempt to
+attach to. You must be :ref:`able to attach a debugger <ptrace privs>` to that
+process, and that process must be a Python process capable of running ``import
+memray`` (that is, the Memray package must be installed in the environment that
+the process is running in).
+
+By default this will open a :doc:`live mode TUI <live>` showing the allocations
+that the process performs after you've attached, but you can instead provide
+the name of a capture file to write to with the ``-o`` option, allowing you to
+analyze the captured allocations with any Memray reporter you'd like. You can
+also provide most of the options that ``memray run`` accepts. See :ref:`the
+CLI reference <memray attach CLI reference>` for details.
```

Comment:
Maybe we should clarify the status of the process after detaching because is not the same in `live` mode and in non `live` mode.

This makes me wonder if we should offer a "stop memray" to stop tracking a previously attached process in non live mode. WDYT?